src/YesSql.Provider.Common/SqlBuilder.cs

```diff
@@ -198,6 +207,18 @@ public virtual string ToSqlString()
 
                 sb.Append("SELECT ");
 
+                if (_distinct)
+                {
+                    sb.Append("DISTINCT ");
+
+                    // This hacky bit needs a better solution
+                    if (_order != null && _order[0] != _dialect.QuoteForColumnName("Id"))
+                    {
+                        _select.Add(",");
+                        _select.AddRange(_order.Where(o => o != " DESC"));
```

Comment:
why filter by `DESC` ?

Reply:
Because we can't put 'DESC' into the SELECT clause.

Reply:
Turns out DISTINCT needs to know about a lot of things to work right.
One thought I had is there might need to be a new clause type.  So there would be "SELECT" and there would be "SELECT DISTINCT".  Then the dialects can have a little more control over it.

Reply:
sorry, I thought the string contained both the name and "desc".
That will work, but still brittle as it expects the "strings" to be separate ...

Reply:
or the builder could not have knowledge of distinct, and the caller to the builder would alter the select clause